AdvanceSpecFlowProject/SpecFlowProject/StepDefinitions/ShareSkillStepDefinitions.cs

```diff
@@ -0,0 +1,257 @@
+                    ManageListingsComponentStepsObj.DeleteShareSkill(skill);
+                }
+                //Adding share skill
+                foreach (var skill in skillList)
+                {
+                    HomePageStepsObj.ClickShareSkillButton();
+                    AddShareSkillStepsObj.AddShareSkill(skill);
+                    Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "AddShareSkill"), "Add share skill successful");
+                }
+            }
+            catch (Exception ex)
+            {
+                Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "AddShareSkill"), "Exception in add share skill");
+                Hook.test.Fail(ex.Message);
+                Assert.Fail();
+            }
+        }
+
+        [Then(@"A skill '([^']*)' should get added")]
+        public void ThenASkillShouldGetAdded(string userSkill)
+        {
+            try
+            {
+                //------Validating skill addition----------
+                List<SkillModel> skillList = JsonReader.GetData<SkillModel>(userSkill);
+
+                foreach (var skill in skillList)
+                {
+                    ManageListingsComponentStepsObj.ValidateShareSkillAdded(skill);
+                    Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "AddShareSkill"), "Skill added successfully validated");
+                }
+            }
+            catch (Exception ex)
+            {
+                Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "AddShareSkill"), "Exception in add skill validation");
+                Hook.test.Fail(ex.Message);
+                Assert.Fail();
+            }
+        }
+
+
+        [When(@"I add invalid skill '([^']*)'")]
+        public void WhenIAddInvalidSkill(string userSkill)
+        {
+            //--Adding invalid skill--
+            try
+            {
+                List<SkillModel> skillList = JsonReader.GetData<SkillModel>(userSkill);
+
+                foreach (var skill in skillList)
+                {
+                    HomePageStepsObj.ClickShareSkillButton();
+                    AddShareSkillStepsObj.AddShareSkill(skill);
+                    Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "AddInvalidShareSkill"), "Add invalid share skill successful");
+                    AddShareSkillStepsObj.ValidateInvalidMessage();
+                }
+            }
+            catch (Exception ex)
+            {
+                Hook.test.AddScreenCaptureFromPath(CommonMethods.SaveScreenshot(driver, "AddInvalidShareSkill"), "Exception in add invalid share skill");
+                Hook.test.Fail(ex.Message);
+                Assert.Fail();
+            }
+        }
+
+        [Then(@"Invalid skill '([^']*)' should not get added")]
+        public void ThenInvalidSkillShouldNotGetAdded(string userSkill)
+        {
+            try
```

Comment:
why there need to put try and catch block ? you should not need this

Reply:
in steps you should only see steps which do things, please fix this everywhere

Reply:
Issue resolved. Please review.